clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -310,6 +326,11 @@ std::set<ConstraintVariable *>
       // lvalue that designates an object that is not a bit-field and is not
       // declared with the register storage-class specifier."
       case UO_AddrOf: {
+        std::set<ConstraintVariable *> AddrVs;
+        if (hasPersistentConstraints(UO)) {
+          return getPersistentConstraints(UO, AddrVs);
+        }
```

Comment:
Per my comment above, I think it would be better to group methods together that require/need a persistent constraint variable.

Reply:
Agree. Refectored.